powermax/helper/port_helper.go

```diff
@@ -63,9 +64,13 @@ func PortDetailMapper(ctx context.Context, port *powermax.DirectorPort) (models.
 	err := CopyFields(ctx, port.SymmetrixPort, &model)
 	model.DirectorID = types.StringValue(port.SymmetrixPort.SymmetrixPortKey.DirectorId)
 	model.PortID = types.StringValue(port.SymmetrixPort.SymmetrixPortKey.PortId)
-	model.NetworkID = types.Int64Value(*port.SymmetrixPort.NetworkId)
-	tpc := int64(*port.SymmetrixPort.TcpPort)
-	model.TPCPort = types.Int64Value(tpc)
+	if port.SymmetrixPort.NetworkId != nil {
```

Comment:
@doriac11 you can use functions like GetNetworkIdOk or HasNetworkId. They are safer.